File: application/src/main/java/com/megustav/lolesports/schedule/processor/MessageProcessor.java
package com.megustav.lolesports.schedule.processor;

import org.telegram.telegrambots.api.methods.BotApiMethod;
import org.telegram.telegrambots.api.objects.Message;

/**
 * Interface defining message processing contract
 *
 * @author MeGustav
 *         15/02/2018 22:26
 */
public interface MessageProcessor {

    /**
     * Process incoming message and form response
     *
     * @param processingInfo enough data to process a message
     * @return formed response
     */
    BotApiMethod<Message> processIncomingMessage(ProcessingInfo processingInfo);

    /**
     * @return processor type
     */
    ProcessorType getType();

}

File: application/src/main/kotlin/com/megustav/lolesports/schedule/processor/MessageProcessor.kt
package com.megustav.lolesports.schedule.processor

import org.telegram.telegrambots.meta.api.methods.BotApiMethod
import org.telegram.telegrambots.meta.api.objects.Message

/**
 * Interface defining message processing contract
 */
interface MessageProcessor {

    val type: ProcessorType

    /**
     * Process incoming message and form response
     *
     * @param processingInfo enough data to process a message
     * @return formed response
     */
    fun processIncomingMessage(processingInfo: ProcessingInfo): BotApiMethod<Message>

}